### DISCUSSION TITLE
Supported Units of System Defined Field

### DISCUSSION THREAD

**Discussion by cllde8** [LabIMotion/labimotion#13] [Announcements]
Check our Wiki page [Supported Units of System-Defined field](https://github.com/LabIMotion/labimotion/wiki/Supported-Units-for-System-Defined-Field)

## 🍎I cannot find the specific units I need, what should I do?

If you cannot find the specific units you require, please add them to this post and provide the information as follows:

- Application Field:
- The unit you are seeking:
- Description of how you intend to use it in your application:

**Comment by medemander** [comment]
Hi, 
I'm looking into adapting Chemotion for TEM, electron diffraction and follow up chemistry for my research group. I have the following unitrequests:

Application field: Voltage
unit: kV
description: used to log the acceleration voltage of a transmission electron microscope

Application field: Percentage
unit: - [dimensionless]
description: input as a decimal number between 0 and 1 since many programs have an output in this format

Application field: resolution
unit: px, MP (pixel, megapixel)
description: log the image resolution of the diffraction image stack (needed for theoretical calculations aftrwards)

The resolution field is optional since it's possible to just use an integer but it would be a nice to have feature.

Cheers,
Paul

**Reply by cllde8** [reply]
Hi @medemander 

We have added
1. kV for Voltage
2. Resolution

These will be included in the next release. Please feel free to try them out on the [test instance](https://labimotion-stage.ibcs.kit.edu/home).

Regarding the "Percentage", we currently provide a % input field. Could you please specify which element/segment/dataset you are using? It would also be very helpful if you could provide a use case.

Thank you.

**Comment by medemander** [comment]
Hi @cllde8,

thanks for adding the units. The percentage in decimal form is an idea to limit the user input to decimal values between 0 and 1. In my case I have an input field for completness of the electron diffreaction measurement dataset, which the program outputs as a decimal number. It would be nice if I could just copy and past this value. 
It would also be possible to just have an input field for decimal numbers, but the only way to do this which I found is with one input group field where "number" is selected. Is there any other input field that can handle decimal number (especially <1)?

**Reply by cllde8** [reply]
Hi @medemander 

Thank you for your reply. We recently received a similar request, which has already been recorded. Would it be okay if I mention you in this ticket as well: https://github.com/LabIMotion/labimotion/issues/56 ?

**Reply by medemander** [reply]
Thank you and yes, you can do that

**Comment by medemander** [comment]
I need two more units added please:

Application field: Pressure
Units: kPa and hPa (Kilo-/Hectopascal)
Usage: common units to log the atmospheric pressure

Application field: Volumes
Units: m³ and cm³

**Reply by cllde8** [reply]
Hi @medemander 

The current “Volumes” setting is litre-based. We are considering creating a new “Volumes (Metric-based)” option to accommodate your request. What do you think?

**Reply by medemander** [reply]
Hi @cllde8,

a seperate option is a great idea, otherwise there will be a lot of different options to cycle through 👍

**Reply by cllde8** [reply]
Hi @medemander 

Thank you for your prompt response. I have a few follow-up questions:

1. Do you need mm³ and µm³ for your application (I assume it’s TEM, or is it related to another context)?
2. Which unit would be more appropriate as the default - m³, cm³, or, if applicable, one of the smaller units mentioned in question 1?

Could you also provide a brief description for the “Volumes (Metric-based)” ?

Thank you.

**Reply by medemander** [reply]
I think some people would appreciate mm³, not sure if µm³ is used commonly, at least I don't need it. I think cm³ would be the better default unit since experiments/analyses in the lab rarely involve chemicals in cubic meter quantities.

**Reply by cllde8** [reply]
Hi @medemander

We have added

1. kPa and hPa for Pressure
2. Volumes (Metric-based) - cm³, mm³, m³

These will be included in the next release. Please feel free to try them out on the [test instance](https://labimotion-stage.ibcs.kit.edu/home) and share your feedback with us. 

Thank you.

**Comment by cllde8** [comment]
A request was received to support the following units for electrochemistry:

```
*Flächengewicht/Grammatur/Mass loading: g cm-2; mg cm-2
*Temparature ramp: °C min-1; °C s-1; °C h-1; K min-1; K s-1; K h-1 
*Tensile force: N or N cm-2
*Percentage: wt.%; vol.%
*Conductivity: mS cm-1
*Molar Conductivity: S cm2 mol-1
*Pressure: kg m-2; t cm-2; etc.
```

The following additions and changes are planned for the next release. We welcome your feedback.

|  | is    | value       | note       |
| ------ | ----- | ------- | ------- |
| Mass loading | new | g cm-2; mg cm-2 | [1] |
| Areal density | new | kg m-2; t m-2; kg cm-2; | [1] |
| Heating rate (Temperature ramp) | add | °C min-1; °C s-1; °C h-1; K min-1; K s-1; K h-1 | [2] |
| Tensile force | new | N; kN; mN | [3] |
| Tensile stress | new | N cm-2; N mm-2 | [3] |
| Mass fraction | new | wt.% | [4] |
| Volume fraction | new | vol.% | [4] |
| Conductivity | add | mS cm-1 |  |
| Molar Conductivity | new | S cm2 mol-1 |  |


[1] The requested units “g cm⁻²; mg cm⁻²” (Mass loading) and “kg m⁻²; t cm⁻²” (originally requested under Pressure) are both defined per unit area. They are provided as two separate unit categories, "Mass loading" and "Areal density", to better reflect their typical scientific usage. The existing Pressure units (Pa, bar, atm, etc.) remain unchanged.

[2] The requested "Temperature ramp" units (°C min⁻¹, °C s⁻¹, °C h⁻¹, K min⁻¹, K s⁻¹, K h⁻¹) have been added into the existing "Heating rate" field. Both terms describe the same physical quantity (temperature change per unit time), and are therefore consolidated into a single field for consistency and to avoid duplication. 


[3] "Tensile force" and "Tensile stress" have been introduced in response to the request for "N or N cm⁻²" (originally listed under Tensile force), to keep force and stress definitions consistent and distinct.

[4] The requested “Percentage” units (wt.% and vol.%) have been implemented as mass "Mass fraction" and "Volume fraction" fields. Both are dimensionless ratios expressed in percent form and are kept as separate unit options to reflect their different physical meanings.

**Comment by herrdivad** [comment]
Hi,

- for Areal density or Mass loading: in paper and fabric industries, **g m<sup>-2</sup>**, is used
- currency like € and $ is also not convertible -> maybe add to hover over?
- Speed: is missing **m/s and km/h**
- formular fields with units need some hint (like hover over) about the units used for calculation or even better, a unit definition on its own. 
<img width="1132" height="216" alt="grafik" src="https://github.com/user-attachments/assets/d4bbc4c8-a263-4673-8218-45d3530f6ff6" /> , maybe use same logic as for tables: 
<img width="724" height="174" alt="grafik" src="https://github.com/user-attachments/assets/15709a48-a7c8-4c5f-95d1-5eb2c3d53432" />

**Comment by MK9876988** [comment]
Hi, 

Since both “Area Density” and “Mass loading” are essentially the same, is it worth combining them? 
--> In the battery research mg/cm^2 as well as g/cm^2 is used. 
The units for energy currently do not include common variants for electrical energy (Ws, kWh); would it be worth adding these?

**Comment by ymkoyama** [comment]
Hello,

I would like to request the addition of MPa (megapascal) to the Pressure system-defined field.

Application Field: Pressure
Requested Unit: MPa (megapascal)
Use case: Materials processing under applied pressure

MPa is commonly used to describe applied pressure in materials processing and engineering. Adding this unit would make it easier to record experimental conditions using commonly reported units.

Thank you.

**Reply by cllde8** [reply]
Dear @ymkoyama ,

Thank you for the suggestion. MPa is a useful addition, and we will add it to the existing Pressure system-defined field.
